qlkube/src/decorateSubscription.js

```diff
@@ -0,0 +1,103 @@
+const { gql } = require('apollo-server-core');
+const { extendSchema } = require('graphql/utilities');
+const { addResolversToSchema } = require('@graphql-tools/schema');
+const { PubSub, withFilter } = require('apollo-server');
+const { capitalizeType } = require('./genericFunction.js');
+
+const pubsub = new PubSub();
+
+function decorateSubscription(
+  baseSchema,
+  targetType,
+  subscriptionField,
+  label
+) {
+  if (!baseSchema) throw 'Parameter baseSchema cannot be empty!';
+  if (!targetType) throw 'Parameter targetType cannot be empty!';
+  if (!subscriptionField) throw 'Parameter subscriptionFiled cannot be empty!';
+  if (!label) throw 'Parameter label cannot be empty!';
+
+  if (baseSchema.getQueryType().getFields()[targetType] === undefined)
+    throw 'Target type not found into the schema';
+
+  let subType =
+    baseSchema._typeMap.Subscription === undefined
+      ? 'type Subscription'
+      : 'extend type Subscription';
+
+  const subscriptionType = capitalizeType(subscriptionField);
+  targetType = capitalizeType(targetType);
+
+  let extension = gql`
+  type  ${subscriptionType} {
+    updateType: String
+    payload: ${targetType}
+  }
+    ${subType} {
+      ${subscriptionField}(name: String, namespace: String!): ${subscriptionType}
+    }
+  `;
+
+  const resolvers = {
+    Subscription: {
+      [subscriptionField]: {
+        subscribe: withFilter(
+          () => pubsub.asyncIterator([label]),
+          (payload, variables, info, context) => {
+            return variables.name === undefined
+              ? true
+              : payload.apiObj.metadata.name === variables.name &&
+                  payload.apiObj.metadata.namespace === variables.namespace;
```

Comment:
The namespaces should always match, this should fix it:
```suggestion
            return payload.apiObj.metadata.namespace === variables.namespace && (variables.name === undefined || payload.apiObj.metadata.name === variables.name);
```

Reply:
yes, you are right!👍